core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/rpc/controlcommands.proto

```diff
@@ -0,0 +1,251 @@
+import "scalapb/scalapb.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/any.proto";
+
+option (scalapb.options) = {
+  scope: FILE,
+  preserve_unknown_fields: false
+  no_default_values_in_constructor: true
+};
+
+message ControlRequest {
+  oneof sealed_value {
+    PropagateChannelMarkerRequest propagateChannelMarkerRequest = 1;
+    TakeGlobalCheckpointRequest takeGlobalCheckpointRequest = 2;
+    DebugCommandRequest debugCommandRequest = 3;
+    EvaluatePythonExpressionRequest evaluatePythonExpressionRequest = 4;
+    ModifyLogicRequest modifyLogicRequest = 5;
+    RetryWorkflowRequest retryWorkflowRequest = 6;
+    ConsoleMessageTriggeredRequest consoleMessageTriggeredRequest = 8;
+    PortCompletedRequest portCompletedRequest = 9;
+    WorkerStateUpdatedRequest workerStateUpdatedRequest = 13;
+    LinkWorkersRequest linkWorkersRequest = 15;
+    Ping ping = 16;
+    Pong pong = 17;
+    Nested nested = 18;
+    Pass pass = 19;
+    ErrorCommand errorCommand = 20;
+    Recursion recursion = 21;
+    Collect collect = 22;
+    GenerateNumber generateNumber = 23;
+    MultiCall multiCall = 24;
+    Chain chain = 25;
+    AddInputChannelRequest addInputChannelRequest = 26;
+    AddPartitioningRequest addPartitioningRequest = 27;
+    AssignPortRequest assignPortRequest = 28;
+    FinalizeCheckpointRequest finalizeCheckpointRequest = 29;
+    InitializeExecutorRequest initializeExecutorRequest = 30;
+    UpdateExecutorRequest updateExecutorRequest = 31;
+    EmptyRequest emptyRequest = 33;
+    PrepareCheckpointRequest prepareCheckpointRequest = 34;
+    QueryStatisticsRequest queryStatisticsRequest = 35;
+  }
+}
+
+message EmptyRequest{}
+
+message AsyncRPCContext {
+  option (scalapb.message).no_box = true;
+  common.ActorVirtualIdentity sender = 1 [(scalapb.field).no_box = true];
+  common.ActorVirtualIdentity receiver = 2 [(scalapb.field).no_box = true];
+}
+
+message ControlInvocation {
+  option (scalapb.message).extends = "edu.uci.ics.amber.engine.common.ambermessage.ControlPayload";
+  string methodName = 1;
+  ControlRequest command = 2 [(scalapb.field).no_box = true];
+  AsyncRPCContext context = 3;
+  int64 commandId = 4;
+}
+
+// Enum for ChannelMarkerType
+enum ChannelMarkerType {
+  REQUIRE_ALIGNMENT = 0;
+  NO_ALIGNMENT = 1;
+}
+
+// Message for ChannelMarkerPayload
+message ChannelMarkerPayload {
+  option (scalapb.message).extends = "edu.uci.ics.amber.engine.common.ambermessage.WorkflowFIFOMessagePayload";
+  common.ChannelMarkerIdentity id = 1 [(scalapb.field).no_box = true];
+  ChannelMarkerType markerType = 2;
+  repeated common.ChannelIdentity scope = 3;
+  map<string, ControlInvocation> commandMapping = 4;
+}
+
+message PropagateChannelMarkerRequest {
+  repeated common.PhysicalOpIdentity sourceOpToStartProp = 1;
+  common.ChannelMarkerIdentity id = 2 [(scalapb.field).no_box = true];
+  ChannelMarkerType markerType = 3;
```

Comment:
can this by None? maybe add [(scalapb.field).no_box = true]

Reply:
the generated code is not an Option because this is an enum. No need to add `no_box`